### ISSUE TITLE
[openbmc] bmcdiscover report unreasonable error message related 2200 port

### ISSUE THREAD

**Issue by robin2008** [xcat2/xcat-core#3207]
I am using Version 2.13.4 (git commit 6ee3741498768994e4bb10d2a77c9699bcabde90, built Tue May 16 10:03:13 EDT 2017)
This is a Management Node
cfgloc=Pg:dbname=xcatdb;host=10.10.1.3|xcatadm
dbengine=Pg

I am using xCAT 2.13.4 dbengine=Pg. I have my dhcp range as 10.80.1.150-250 in my networks table. When i run bmcdiscover I get
the following output  [root@mn1 ~]# bmcdiscover --range 10.10.1.150-250
10.10.1.152,8335-GCA,7817C0A,,,mp,bmc
10.10.1.153,8335-GCA,7817BEA,,,mp,bmc
10.10.1.154,8335-GCA,7817BFA,,,mp,bmc
Error: bmcdiscover plugin bug, pid 24702, process description: 'xcatd SSL: bmcdiscover for root@localhost: bmcdiscover instance' with error 'Can't use string ("Can not get status of 2200 port.") as a HASH ref while "strict refs" in use at /opt/xcat/lib/perl/xCAT/MsgUtils.pm line 407.

It runs ok when I run bmcdiscover --range 10.10.1.0/24.  Any ideas why this plugin bug show up?

**Comment by robin2008** [comment]
@xuweibj Could you help to take a look, it seems some error cases(openbmc related discovery) are not handled well in our codes.

**Comment by whowutwut** [comment]
This was reported by customer. We need to add a testcase to catch things like this. It's in normal path for IPMI based open power machines

**Comment by robin2008** [comment]
I think we already have case to check the bmcdiscover return code,  need to see if it could cover more error cases. @junxiawang could take a look 

But for this ticket, just confirmed with @xuweibj, it seems we didn't know how to mimic such customer error case (port 2200 status is not found in nmap output).

**Comment by cxhong** [comment]
I asked them for nmap output this morning.   the error case couldn't reproduce any more.  bmcdiscover command returns back with good results today.  So not sure what kind of corner case they hit yesterday.
````
[root@mn1 ~]# nmap 10.10.1.152-250 -p 2200 -Pn

Starting Nmap 6.40 ( http://nmap.org ) at 2017-06-06 09:17 CDT
Nmap scan report for 10.10.1.152
Host is up (0.00032s latency).
PORT     STATE  SERVICE
2200/tcp closed ici
MAC Address: 70:E2:84:14:08:4B (Wistron InfoComm(Zhongshan))

Nmap scan report for 10.10.1.153
Host is up (0.00022s latency).
PORT     STATE  SERVICE
2200/tcp closed ici
MAC Address: 70:E2:84:14:07:6A (Wistron InfoComm(Zhongshan))

Nmap scan report for 10.10.1.154
Host is up (0.00033s latency).
PORT     STATE  SERVICE
2200/tcp closed ici
MAC Address: 70:E2:84:14:08:E9 (Wistron InfoComm(Zhongshan))

Nmap scan report for 10.10.1.156
Host is up (0.00033s latency).
PORT     STATE  SERVICE
2200/tcp closed ici
MAC Address: 70:E2:84:14:17:23 (Wistron InfoComm(Zhongshan))

Nmap scan report for 10.10.1.161
Host is up (0.00024s latency).
PORT     STATE  SERVICE
2200/tcp closed ici
MAC Address: CC:37:AB:B2:31:C0 (Unknown)

Nmap scan report for cumulus.handlogten.net (10.10.1.162)
Host is up (0.00024s latency).
PORT     STATE  SERVICE
2200/tcp closed ici
MAC Address: CC:37:AB:B2:33:80 (Unknown)

Nmap scan report for 10.10.1.163
Host is up (0.00018s latency).
PORT     STATE  SERVICE
2200/tcp closed ici
MAC Address: 8C:EA:1B:26:8E:00 (Unknown)

Nmap done: 99 IP addresses (7 hosts up) scanned in 1.33 seconds
````

**Comment by xuweibj** [comment]
The issue was be fixed in pull request #3209.

**Comment by robin2008** [comment]
As this corner case is hard to simulate,  we can only verify it with UT.
